libreoffice-starter-extension-master/source/org/libreoffice/example/dialog/ConfigDialog.java

```diff
@@ -0,0 +1,84 @@
+package org.libreoffice.example.dialog;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.libreoffice.example.helper.DialogHelper;
+
+import com.sun.star.awt.XDialog;
+import com.sun.star.awt.XDialogEventHandler;
+import com.sun.star.awt.XFixedText;
+import com.sun.star.awt.XTextComponent;
+import com.sun.star.lang.WrappedTargetException;
+import com.sun.star.uno.XComponentContext;
+
+public class ConfigDialog implements XDialogEventHandler{
+	private XDialog dialog;
+	private XComponentContext xContext;
+	private String homeFolder = null;
+	private static final String actionCheck = "checkNow";
+	/** String of knownc actions */
+	private String[] supportedActions = new String[] { actionCheck };
+	/** User entered client id */
+	private String id;
+
+	public ConfigDialog (XComponentContext xContext2, String homeFolder) {
+		this.dialog = DialogHelper.createDialog("config_dialog.xdl", xContext2, this);
+		this.xContext = xContext2;
+		this.homeFolder = homeFolder;
+	}
+
+	public void show(){
+		dialog.execute();
+	}
+
+	/**
+	 * When user presses button, input id is checked.
+	 * If it's valid, then it is saved in a file and set to variable.
+	 *   Dialog has ended and user can use translation services.
+	 * If id is not valid, user has to try again in order to
+	 *   translate anything. Info field is set to explain.
+	 */
+	private void onCheckButtonPressed() {
+		XTextComponent idField = DialogHelper.getEditField( this.dialog, "clientIDField" );
+		id = idField.getText();
+		boolean valid = ConfigID.checkID(id);
+		if(valid) {
+			try {
+				FileWriter dataFile = new FileWriter(homeFolder + File.separator +"tildeID");
```

Comment:
Client-ID saglabāšanu failā labāk implementēt `ConfigID` klasē, lai ir viena klase atbildīga gan par to kā nolasīt un arī saglabāt atslēgu. Ja mums būs jāmaina veids kā saglabāt atslēgu, tad būs jāveic izmaiņas tikai vienā klasē. Piemēram mēs potenciāli varētu ieviest atslēgas šifrēšanu vai glabāt atslēgu citur un tas tad būtu vienkāršāk maināms vienā klasē.